### PR TITLE
Pythnize/cs properties

### PR DESCRIPTION
**Pull request by SengerM**
Interface is more robust and user friendly. Did it only for some functions I am using.

### CODE REVIEW

File: python/CSXCAD/CSProperties.pyx [hunk 1]
```diff
@@ -388,16 +389,30 @@ cdef class CSProperties:
...
+            An object of type `CSPrimitives` with the primitive just created.
 
         See Also
         --------
         CSXCAD.CSPrimitives.CSPrimPolyhedronReader : See here for details on primitive arguments
         """
-        return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRONREADER, filename=filename, **kw)
+        file = Path(file) # Make sure that whatever we receive can be interpreted as a path to a file.
+        if not file.is_file():
+            raise FileNotFoundError(f'`file` points to a file that does not exist. Received {repr(str(file))}, absolute path is {repr(str(file.resolve()))}. ')
```
Comment (thliebig) [inline python/CSXCAD/CSProperties.pyx:411]: Does it actually matter here if the file exists or not? For example we just define the file name, it is then later created and then maybe read? Why check the existence here if it does not need to exist (yet)? 
Again this feels unnecessary restrictive?

File: python/CSXCAD/CSProperties.pyx [hunk 2]
```diff
@@ -388,16 +389,30 @@ cdef class CSProperties:
         """
         return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRON, **kw)
 
-    def AddPolyhedronReader(self, filename, **kw):
-        """ AddPolyhedronReader(filename, **kw)
+    def AddPolyhedronReader(self, file:Path, **kw)->CSPrimitives:
```
Comment (thliebig) [inline python/CSXCAD/CSProperties.pyx:392]: I prefer to supply str? why not make this the default type? or at least both?

File: python/CSXCAD/CSProperties.pyx [hunk 3]
```diff
@@ -452,25 +467,39 @@ cdef class CSPropMaterial(CSProperties):
...
+        # Do the same as in the previous implementation:
+        kw = dict(epsilon=epsilon, mue=mue, kappa=kappa, sigma=sigma, density=density)
+        for prop_name,val in kw.items():
+            if val is None:
+                continue
             if prop_name == 'density':
                 (<_CSPropMaterial*>self.thisptr).SetDensity(val)
                 continue
-            if type(val)==float or type(val)==int:
+            if isinstance(val, (int,float)):
```
Comment (thliebig) [inline python/CSXCAD/CSProperties.pyx:500]: It is better as it was but again a bit restrictive?  We should simply test it is an iterable and if not try to convert to float instead?

File: python/CSXCAD/CSProperties.pyx [hunk 4]
```diff
@@ -452,25 +467,39 @@ cdef class CSPropMaterial(CSProperties):
...
+            if val is not None:
+                if isinstance(val, (int,float)):
+                    continue
+                elif  isinstance(val, (tuple,list)) and len(val)==3 and all([isinstance(_, (int,float)) for _ in val]):
+                    continue
+                else:
+                    raise ValueError(f'`{name}` must be a scalar or a 3-vector of scalars, received `{val}`. ')
+        # Validate density:
+        if density is not None:
+            if not isinstance(density, (int, float)):
```
Comment (thliebig) [inline python/CSXCAD/CSProperties.pyx:490]: better to try to convert to float? What happens with e.g. np.float ?

File: python/CSXCAD/CSProperties.pyx [hunk 5]
```diff
@@ -388,16 +389,30 @@ cdef class CSProperties:
...
         See Also
         --------
         CSXCAD.CSPrimitives.CSPrimPolyhedronReader : See here for details on primitive arguments
         """
-        return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRONREADER, filename=filename, **kw)
+        file = Path(file) # Make sure that whatever we receive can be interpreted as a path to a file.
+        if not file.is_file():
+            raise FileNotFoundError(f'`file` points to a file that does not exist. Received {repr(str(file))}, absolute path is {repr(str(file.resolve()))}. ')
+
+        file = file.resolve() # Convert the path into an absolute path, otherwise it fails silently later on.
```
Comment (thliebig) [inline python/CSXCAD/CSProperties.pyx:413]: I'm not sure I like this. In some case I want to be relative (to the project) and not absolute? 
Again I do not like to take options from the user?

File: python/CSXCAD/CSProperties.pyx (hunk 1 above)
Comment (SengerM) [reply python/CSXCAD/CSProperties.pyx:411]: I think that a [fail-fast approach](https://en.wikipedia.org/wiki/Fail-fast_system) makes it easy for the user. Are there many cases in which the file would be created after the configuration is done? Not an expert here, but it sounds to me that the answer is no.

File: python/CSXCAD/CSProperties.pyx (hunk 3 above)
Comment (SengerM) [reply python/CSXCAD/CSProperties.pyx:500]: Isn't it the same as before my change, but only written more compact? In both cases you make sure `val` is either `float` or `int`.

File: python/CSXCAD/CSProperties.pyx (hunk 2 above)
Comment (SengerM) [reply python/CSXCAD/CSProperties.pyx]: Done.

File: python/CSXCAD/CSProperties.pyx (hunk 5 above)
Comment (SengerM) [reply python/CSXCAD/CSProperties.pyx:413]: The user can still provide a relative path to the argument `file`.

**Pull request closed without merging**